readthedocs/api/v2/views/footer_views.py

```diff
@@ -142,13 +144,12 @@ def _get_context(self):
         version = self._get_version()
 
         page_slug = self.request.GET.get('page', '')
+        path = ''
         if page_slug and page_slug != 'index':
-            if main_project.documentation_type == 'sphinx_htmldir':
-                path = page_slug + '/'
+            if version.documentation_type == 'sphinx_htmldir':
+                path = re.sub('/index$', '', page_slug) + '/'
```

Comment:
Shouldn't we use the regex module that has timeout here?

Reply:
regex are dangerous when the user provides the pattern, this pattern is simple enough.